Getting Started with Synthetic Monitoring > Multistep API Tests (#10946)
DataDog/documentation: content/en/getting_started/synthetics/api_test.md

When you're ready to run your test, click **Save Test**. 

## Create a multistep API test

> Multistep API tests also **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are responding to HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 

Reviewer: ```suggestion
Multistep API tests **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are responding to HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 
```

> Multistep API tests also **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are responding to HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 

Author: ```suggestion
Multistep API tests also **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are executing HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 
```

> Multistep API tests also **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are responding to HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 

Reviewer: +1

> Multistep API tests also **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are responding to HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 

Author: How about this? @kayayarai @sarina-dd

> Multistep API tests also **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are responding to HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 

Reviewer 2: ```suggestion
Multistep API tests chain multiple HTTP requests together, allowing you to configure API tests on complex backend systems. 
```
I think we could whittle it down to this ^

> Multistep API tests also **monitor your API endpoints** and **alert you** when they become too slow or fail. These tests verify that your services are responding to HTTP requests and meeting any conditions you define, such as **response time**, **HTTP status code**, and **header or body contents**. 

Author: yas

### Configure the test

> Creating a multistep API test slightly differs from creating an API test. You can link HTTP requests and create variables from the request response data in **Define steps**. 

Author: Is "aPI" a typo here @kayayarai?

> Creating a multistep API test slightly differs from creating an API test. You can link HTTP requests and create variables from the request response data in **Define steps**. 

Reviewer 2: Yes _shakes fist at wireless keyboard_

1. In the Datadog application, hover over **[UX Monitoring][2]** in the left hand menu and select **Synthetic Test**.
2. In the top right corner, click the **New Test** button.
3. Select **Multistep API test**.
4. Define your multistep API request like you would with an API test.
5. In **Define steps**, click **Create Your First Step**. 

    - Add a name to your test HTTP request step.
    - Specify the URL you want to query. 

>     - You can select **Advanced Options** to add request options, request body, certificates, and proxies.

Reviewer: ```suggestion
    -  Select **Advanced Options** to add request options, request body, certificates, and proxies.
```

    - Click **Test URL**. 

>     - You can optionally add new assertions, enable execution parameters, and extract variables from the response content.

Reviewer: ```suggestion
    - Optionally, add new assertions, enable execution parameters, and extract variables from the response content.
```

    - Click **Save Step**.

6. Click **Add Another Step** to add another step. You can create up to ten steps. 
7. Configure the rest of your test conditions including test frequency, alerting conditions, and alert message.

> When you're ready to run your test, click **Save Test**. 

Reviewer: ```suggestion
When you're ready to create your test, click **Save Test**. 
```

For more information, see [Multistep API Tests][7].

## Test results

The API test and multistep API test detail pages display an overview of the test configuration, the global uptime associated with the tested endpoint by location, graphs about response time and network timings, and a list of test results and events.

To troubleshoot a failed test, scroll down to **Test Results** and click on a failing test result. Review failed assertions and response details such as status code, response time, and associated headers and body to diagnose the issue.

{{< img src="getting_started/synthetics/api-test-failure.png" alt="API test failure"  style="width:90%;">}}

> With Datadog's [APM integration with Synthetic Monitoring][5], you can access the root cause of a failed test run by looking at the trace generated from the test run in the **Traces** tab.

Reviewer: ```suggestion
With Datadog's [APM integration with Synthetic Monitoring][5], access the root cause of a failed test run by looking at the trace generated from the test run in the **Traces** tab.
```
